File: easy/1/main.go
package main

import "fmt"

func main() {
	fmt.Println(twoSum([]int{2, 7, 11, 15}, 9))
	fmt.Println(twoSum([]int{3, 2, 4}, 6))
	fmt.Println(twoSum([]int{3, 3}, 6))
}

func twoSum(nums []int, target int) []int {
	for i, num := range nums {
		for j, num2 := range nums[i+1:] {
			if num+num2 == target {
				return []int{i, i + j + 1}
			}
		}
	}
	return []int{0, 0}
}

File: easy/1/main.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {

    // create a hash map
    auto nums_map = unordered_map<int, int>();

    for (auto i = 0; i < nums.size(); i++) {
      auto finded_index = nums_map.find(target - nums[i]);

      if (finded_index != nums_map.end()) {
        return {finded_index->second, i};
      } else {
        nums_map[nums[i]] = i;
      }
    }
    return {};
  }
};
